src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -36,26 +41,34 @@ class Lambda {
   private val EmptyJsonBody: RequestBody =
     RequestBody.create(MediaType.parse("application/json; charset=utf-8"), "")
 
+  private sealed trait PurgeType
+  private object Soft extends PurgeType
+  private object Hard extends PurgeType
```

Comment:
You could override `toString` in these two if you wanted slightly more eye-pleasing logging:

```scala
private object Soft extends PurgeType { override def toString = "soft" }
// ...
println(s"Sent $purgeType purge request")
```